src/ec2/model.ts

```diff
@@ -113,19 +128,29 @@ export class Ec2ConnectionManager {
         const hasPermission = await this.hasProperPermissions(IamRole!.Arn)
 
         if (!hasPermission) {
-            const message = `Ensure an IAM role with the proper permissions is attached to the instance. Found attached role: ${
-                IamRole!.Arn
-            }`
-            this.throwConnectionError(message, selection, {
-                code: 'EC2SSMPermission',
-                documentationUri: this.policyDocumentationUri,
-            })
+            const policiesAdded = await promptToAddPolicies(this.iamClient, IamRole!.Arn!)
+            if (!policiesAdded) {
+                const message = `Did not add permissions. Ensure an IAM role with the proper permissions is attached to the instance. Found attached role: ${
+                    IamRole!.Arn
+                }`
+                this.throwConnectionError(message, selection, {
+                    code: 'EC2SSMPermission',
+                    documentationUri: this.policyDocumentationUri,
+                })
+            } else {
+                const timeout = new Timeout(policyAttachDelay)
+
+                function delay(ms: number) {
+                    return new Promise(resolve => setTimeout(resolve, ms))
+                }
+                await showMessageWithCancel(`Adding Inline Policy to ${IamRole!.Arn}`, timeout)
```

Comment:
this is a strange place for this logic. Where is the code that actually adds the policy?

Reply:
The logic for adding a policy is on line 131 just above with a call to `promptToAddPolicies` declared in `src/shared/remoteSession.ts`.  We prompt the user to see if they want to add the policy, then handle the response here. I moved the `promptToAddPolicies` to `src/shared/remoteSession.ts` since it seemed like a general utility function for establishing SSM connection, and not specific to EC2.

Reply:
should the "Adding policy..." progress message start before the call to `addSsmActionsToInlinePolicy` ?

Reply:
I refactored the delay logic to be in the `remoteSession.ts` file to make this change.